chain/txValidator.go

```diff
@@ -167,13 +169,24 @@ func CheckTransactionPayload(txn *transaction.Transaction) error {
 		}
 
 		// Check sub.Meta & sub.Identifier limitation via height
-		h := DefaultLedger.Store.GetHeight() + 1 // txn's height should be currHeight + 1
 		if len(pld.Identifier) > config.MaxTxnSubIdentifierList.GetValueAtHeight(h) {
 			return errors.New("Identifier too long")
 		}
 		if len(pld.Meta) > config.MaxTxnSubMetaList.GetValueAtHeight(h) {
 			return errors.New("Meta too long")
 		}
+	case pb.UNSUBSCRIBE_TYPE:
+		pld := payload.(*pb.Unsubscribe)
+
+		topic := pld.Topic
+		match, err := regexp.MatchString("(^[A-Za-z][A-Za-z0-9-_.+]{2,254}$)", topic)
```

Comment:
Do we need to check topic for unsubscribe at all? It should be enough if we simply check whether an existing subscription exist right?

Reply:
existing subscription is gonna be checked later, doing leveldb search
if we can discard it earlier and with less work I think it could be worth it
but prob should create some common function to check topic name for both sub and unsub

Reply:
Yeah agree

Reply:
Speaking about valid names, I think `/` and `*` should be allowed in the name. Using that you can make structures like subscription: `games/spacecats/matchmaking`, `games/spacecats/leaderboards`, `games/spacecats/*`
What do you think?

Reply:
I think we should probably discuss this out of scope of this PR